src/compas_viewer/ui/mainwindow.py

```diff
@@ -19,3 +22,17 @@ def set_window_central(self) -> None:
         central = QWidget()
         central.setLayout(layout)
         self.setCentralWidget(central)
+
+    def set_window_unit(self) -> None:
+        if self.viewer.config.window.unit == "m":
+            grid_size = (10000, 10, 10000, 10)
+        elif self.viewer.config.window.unit == "cm":
+            grid_size = (100, 10, 100, 10)
+        elif self.viewer.config.window.unit == "mm":
+            grid_size = (10, 10, 10, 10)
+        else:
+            print("Invalid unit")
+            return
```

Comment:
i would prefer to deal wit units in a separate PR because it should be related to `compas.tolerance`

Reply:
okay I will just focus on resole `gridobject` here